src/unix/core.c

```diff
@@ -1648,3 +1649,35 @@ int uv__search_path(const char* prog, char* buf, size_t* buflen) {
   /* Out of tokens (path entries), and no match found */
   return UV_EINVAL;
 }
+
+
+unsigned int uv_available_parallelism(void) {
+#ifdef __linux__
+  cpu_set_t set;
+  long rc;
+
+  memset(&set, 0, sizeof(set));
+
+  /* sysconf(_SC_NPROCESSORS_ONLN) in musl calls sched_getaffinity() but in
+   * glibc it's... complicated... so for consistency try sched_getaffinity()
+   * before falling back to sysconf(_SC_NPROCESSORS_ONLN).
+   */
+  if (0 == sched_getaffinity(0, sizeof(set), &set))
```

Comment:
Might be better to use the same as the existing API (v2) here? https://github.com/libuv/libuv/blob/853f66f0b14a7bcc5b901ab465f0e12b634b3729/src/unix/thread.c#L312-L316

Reply:
I don't think that's necessary because the code in this PR is only interested in the current thread's affinity, not some other thread.